### PR TITLE
23 add quill

### LINKED ISSUE #23: Add quill
Use react quill as text editor

### PR DESCRIPTION
**Pull request by 20jasper**
Add a very minimal Quill implementation. We can add more rich text editing if we want later on

closes #23 

A very basic multiline text editor with example text

### CODE REVIEW

File: app/components/TextEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import dynamic from "next/dynamic";
+// we must disable SSR since ReactQuill attempts to access the `document`
+const ReactQuill = dynamic(() => import('react-quill'), { ssr: false });
```
Comment (alcpereira) [inline app/components/TextEditor.tsx:3]: Adding a loading state would be nice here.

File: app/page.tsx [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
   const inputValue = useQuery(api.input.getInputValue);
 
   const mutateValue = useMutation(api.input.updateInputValue);
 
-  const handleOnChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const payload = {
-      id: event.target.attributes.getNamedItem("hidden-id")?.value as Id<"input">,
-      text: event.target.value
+  const handleOnChange: OnChange = (_value, _delta, _source, editor) => {
+    if (inputValue === undefined || inputValue === null) {
```
Comment (alcpereira) [inline app/page.tsx:16]: We will need to address this at some point (but not in this PR) 😅 

Probably in #22

**Review by alcpereira — DISMISSED** [review]

File: app/components/TextEditor.tsx (hunk 1 above)
Comment (alcpereira) [inline app/components/TextEditor.tsx:3]: /reviewpad explain

File: app/page.tsx (hunk 2 above)
Comment (20jasper) [reply app/page.tsx:16]: I just added [#28](https://github.com/web-dev-cody-hackathon/100editors/issues/28) for this issue

**Review by pereiraA-ppb — APPROVED** [review]

**Review by alcpereira — APPROVED** [review]

**Pull request merged**